gfx/drivers_context/wayland_ctx.c

```diff
@@ -571,17 +574,55 @@ static const struct xdg_surface_listener xdg_surface_listener = {
 };
 
 static void handle_toplevel_config(void *data, struct xdg_toplevel *toplevel,
-                                   int width, int height, struct wl_array *states)
+                                   int32_t width, int32_t height, struct wl_array *states)
 {
     gfx_ctx_wayland_data_t *wl = (gfx_ctx_wayland_data_t*)data;
     
-    /* TODO: implement resizing */
-    (void)toplevel;
-    
-    wl->width = wl->buffer_scale * width;
-    wl->height = wl->buffer_scale * height;
-    
-    wl->configured = false;
+    printf("Configure event got, width: %d, height: %d\n", width, height);
+    wl->fullscreen = false;
+    wl->maximized = false;
+    enum xdg_toplevel_state *state;
```

Comment:
Same thing here. Not C89 compliant. Declare variables at the top of a function or code block.